utils/error_injection.hh

```diff
@@ -256,6 +338,36 @@ public:
         return make_exception_future<>(exception_factory());
     }
 
+    // \brief Inject exception
+    // \param func function returning a future and taking an injection handler
+    template <typename Func>
+    requires std::is_invocable_r_v<future<>, Func, injection_handler&>
+    future<> inject_with_handler(const std::string_view& name,
+                                 Func&& func) {
+        auto data_opt = get_data(name);
+        if (!data_opt) {
+            co_return;
+        }
+
+        auto& data = data_opt->get();
+
+        bool one_shot = data.is_one_shot();
+        if (one_shot && data.is_ongoing_oneshot()) {
+            co_return;
```

Comment:
Please add a comment which explains why we don't fire an injection even though it's enabled